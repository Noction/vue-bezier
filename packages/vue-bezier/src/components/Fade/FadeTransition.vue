<template>
  <component
    :is="componentType"
    :tag="tag"
    enter-active-class="fade-in"
    move-class="fade-move"
    leave-active-class="fade-out"
    v-bind="{ ...$attrs, ...hooks }"
  >
    <slot />
  </component>
</template>

<script lang="ts">
export default {
  inheritAttrs: false
}
</script>

<script setup lang="ts">
import {
  buildComponentType,
  buildEmits,
  buildHooks,
  buildProps,
  buildTag
} from '../../composable'

const props = defineProps(buildProps())
const emit = defineEmits(buildEmits())

const componentType = buildComponentType(props)
const tag = buildTag(props)
const hooks = buildHooks(props, emit)

</script>

<style>
  @keyframes fade-in {
    from { opacity: 0; }
    to { opacity: 1; }
  }

  @keyframes fade-out {
    from { opacity: 1; }
    to { opacity: 0; }
  }

  .fade-in { animation-name: fade-in; }
  .fade-out { animation-name: fade-out; }
  .fade-move { transition: transform .3s ease-out; }
</style>
