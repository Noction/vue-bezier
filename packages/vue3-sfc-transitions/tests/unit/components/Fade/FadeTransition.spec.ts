import FadeTransition from '../../../../src/components/Fade/FadeTransition.vue'
import { mount } from '@vue/test-utils'

describe('components / Fade / FadeTransition', () => {
  it('When mounting component with default props, it should work fine', () => {
    const wrapper = mount(FadeTransition)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('When mounting component with group enabled prop, it should work fine', () => {
    const wrapper = mount(FadeTransition, { props: { group: true } })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
