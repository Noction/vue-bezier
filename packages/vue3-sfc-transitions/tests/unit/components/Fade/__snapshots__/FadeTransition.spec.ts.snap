// Vitest Snapshot v1

exports[`components / Fade / FadeTransition > When mounting component with default props, it should work fine 1`] = `"<transition-stub></transition-stub>"`;

exports[`components / Fade / FadeTransition > When mounting component with default props, it should work fine 2`] = `"<transition-group-stub></transition-group-stub>"`;

exports[`components / Fade / FadeTransition > When mounting component with group enabled prop, it should work fine 1`] = `"<transition-group-stub></transition-group-stub>"`;
